Toolkit/ArcGISToolkit/BookmarksViewController.swift

```diff
@@ -0,0 +1,171 @@
+//
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+/// The protocol you implement to respond to user bookmark selection.
+/// view controller.
+/// - Since: 100.7.0
+public protocol BookmarksViewControllerDelegate: AnyObject {
+    /// Tells the delegate that the user has selected a bookmark.
+    ///
+    /// - Parameters:
+    ///   - bookmark: The new bookmark selected.
+    /// - Since: 100.7.0
+    func bookmarkSelectionDidChange(_ bookmark: AGSBookmark)
```

Comment:
Since I have an internal `UITableViewController` which actually handles the row selection I make the controller parameter be a `UIViewController`.  Is that OK?

Reply:
Do you mean you're passing an instance of `BookmarksTableViewController`? The passed object should be an instance of `BookmarksViewController`. The user doesn't (or shouldn't) know or care that there's an underlying view controller. Rather than making the type of the parameter `UIViewController` to fit the communication pattern between the parent and child view controller, I would recommend changing that communication pattern. There's lots of options. You could use a closure property on the child or you could create a different delegate type for the child or you could give a child a reference to the parent (or just use the `parent` property and cast to the parent type). Does that make sense?